src/utils/Utils.ts

```diff
@@ -1607,4 +1608,92 @@ export default class Utils {
+      if (!Utils.isEmptyArray(siteAreaTreeNode.siteAreaChildren)) {
+        // Check all children and children of children
+        for (const siteAreaChild of siteAreaTreeNode.siteAreaChildren) {
+          if (siteAreaChild.id === siteAreaID) {
+            return siteAreaTreeNode;
+          }
+          if (this.findSiteAreaInTrees(siteAreaID, null, siteAreaChild)) {
+            return siteAreaTreeNode;
+          }
+        }
+      }
+      return null;
+    }
+    if (Array.isArray(siteAreas)) {
+      // Loop through trees to find specific tree
+      for (const siteArea of siteAreas) {
+        // If site area ID is defined return tree, which contains the site area
+        const requestedTree = this.findSiteAreaInTrees(siteAreaID, null, siteArea);
+        if (requestedTree) {
+          return siteArea;
+        }
+      }
+    }
+  }
+
+  public static buildSiteAreaTrees(siteAreas: SiteArea[]): SiteArea[] {
+    // Hash Table helper
+    const hashTable = {};
+    for (const siteArea of siteAreas) {
+      hashTable[siteArea.id] = { ...siteArea, siteAreaChildren: [] } as SiteArea;
+    }
+    const siteAreaTrees: SiteArea[] = [];
+    // Build tree
+    for (const siteArea of siteAreas) {
+      if (!Utils.isNullOrUndefined(siteArea.parentSiteAreaID) && !Utils.isNullOrUndefined(hashTable[siteArea.parentSiteAreaID])) {
+        // Push sub site area to parent children array
+        hashTable[siteArea.parentSiteAreaID].siteAreaChildren.push(hashTable[siteArea.id]);
+      } else {
+        // If no parent ID is defined push root site area to array
+        siteAreaTrees.push(hashTable[siteArea.id]);
+      }
+    }
+    return siteAreaTrees;
+  }
+
+  public static checkSiteAreaTrees(siteAreaTrees: SiteArea[], numberOfSiteAreas?: number,
+      siteAreaTree: SiteArea = null, count = 0) : number {
+    if (siteAreaTree) {
+      count++;
+      // Count and check all children and children of children
+      if (!Utils.isEmptyArray(siteAreaTree.siteAreaChildren)) {
+        for (const siteAreaChild of siteAreaTree.siteAreaChildren) {
+          if (siteAreaTree.smartCharging !== siteAreaChild.smartCharging ||
+            siteAreaTree.numberOfPhases !== siteAreaChild.numberOfPhases ||
+            siteAreaTree.siteID !== siteAreaChild.siteID ||
+            siteAreaTree.voltage !== siteAreaChild.voltage) {
+            throw new BackendError({
+              siteAreaID: siteAreaChild.id,
+              method: 'checkSiteAreaTrees',
+              message: 'Property inconsistency in Site Area Tree',
+            });
+          }
+          count += this.checkSiteAreaTrees(null, null, siteAreaChild);
+        }
+      }
+      return count;
+    }
+    if (Array.isArray(siteAreaTrees)) {
+      // Loop through trees to check properties and count all site areas in it
+      for (const singleTree of siteAreaTrees) {
+        count += this.checkSiteAreaTrees(null, null, singleTree);
+      }
+      if (count !== numberOfSiteAreas) {
```

Comment:
I would expect (count > numberOfSiteAreas) here and breaks the recursive loop.
How do you break the circular deps in one tree (A > B > C > A)?
I don't know if we have this use case.

Reply:
@LucasBrazi06 
Let me explain real quick:
The reason it works is the building process.
Here you see the build of a consistent tree list:

![image](https://user-images.githubusercontent.com/33114317/151165065-d8b095b9-8856-4103-bee8-f5b761cbe5b1.png)
As you see in the last step the site area without a parent is pushed in the final list which is counted.

If you have a circular dependency the following happens:
![image](https://user-images.githubusercontent.com/33114317/151165223-cd8ec161-1d26-4145-8ae3-42e7e89d3b28.png)
The circular structure is not pushed to the final list, because there is no root. This will result in less site areas in total, then the source input

Reply:
Ok I understand, thanks for the explanation!
You are just checking about having no trees with no root element.